File: app/src/main/java/io/github/ovso/righttoknow/framework/di/AppModule.java
package io.github.ovso.righttoknow.framework.di;

import android.app.Application;
import android.content.Context;
import com.google.android.gms.ads.InterstitialAd;
import dagger.Module;
import dagger.Provides;
import io.github.ovso.righttoknow.framework.ad.MyAdView;

@Module public class AppModule {
  @Provides Context provideContext(Application application) {
    return application;
  }

  @Provides InterstitialAd provideInterstitialAd(Application context) {
    return MyAdView.getInterstitalAd(context);
  }
}

File: app/src/main/java/io/github/ovso/righttoknow/framework/di/AppModule.kt
package io.github.ovso.righttoknow.framework.di

import android.app.Application
import android.content.Context
import com.google.android.gms.ads.InterstitialAd
import dagger.Module
import dagger.Provides
import io.github.ovso.righttoknow.framework.ad.MyAdView.getInterstitalAd

@Module
internal class AppModule {
  @Provides
  fun provideContext(application: Application): Context {
    return application
  }

  @Provides
  fun provideInterstitialAd(context: Application?): InterstitialAd {
    return getInterstitalAd(context)
  }
}